src/Hintman/Suggestion/TrailingSpaces.hs

```diff
@@ -0,0 +1,34 @@
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Hintman.Suggestion.TrailingSpaces
+       ( getTrailLines
+       , toLines
+       , suggest
+       , suggestRaw
+       ) where
+
+import Data.Text (stripEnd)
+import Hintman.Suggestion.Core (Line(..), LineState(..), Suggestion(..))
+
+
+toLines :: Text -> [Line]
+toLines txt = map (\(n,x) -> Line{lineRaw = n, lineBody = x, lineState = lineState' x}) lTxt
+  where
+    lTxt :: [(Int, Text)]
+    lTxt = zip [1..] (lines txt)
+
+    lineState' :: Text -> LineState
+    lineState' x
+        | x == stripEnd x = Plain
+        | otherwise = Trailing
+
+getTrailLines :: [Line] -> [Line]
+getTrailLines = filter (\ Line{lineState} -> lineState == Trailing)
+
+suggest :: FilePath -> [Line] -> [Suggestion]
+suggest path list =
+    map (\Line{..} -> Suggestion {suggestionFile = path, suggestionNew = stripEnd lineBody, suggestionComment = Nothing})
```

Comment:
If my previous comment will be applied you won't need to make `stripEnd` twice 🙂